### DISCUSSION TITLE
Compiling Unreal Engine on KDE-Debian, (GNU/Linux).

### DISCUSSION THREAD

**Discussion by MATHEW2HAPPY** [community/community#60959] [Programming Help]
Hi, I am new to compiling software, although I have some experience with compiling small C++ programs. 

I use a Laptop on which Windows 10 is my host OS, and I run KDE-Debian inside of VMware Workstation Pro 17.  I have Unreal Engine 5 installed on Windows 10, but I would like to compile the source code to run on KDE-Debian.  I have git installed on Debian, along with a GUI for it called GitKraken.  I have also installed the Atom Text Editor.

On the web: I have an Epic Games account, I have a GitHub accout, and I have linked them successfully!  

Now, I am stuck.  I think I found the webpage on GitHub for Unreal Engine, but I can't find the Official Source Code.  All I can find are other peoples forks of the Official Code.  Can someone tell me where it is; I would like an URL to the Source Code.

Thanks,
Mathew

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @MATHEW2HAPPY ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RamtinKosari** [comment]
Hi dear @MATHEW2HAPPY 
To access the official source code of Unreal Engine, you need to follow these steps:

1. Go to the Epic Games GitHub repository for Unreal Engine by visiting the following URL: [https://github.com/EpicGames/UnrealEngine ↗](https://github.com/EpicGames/UnrealEngine).

2. Once you're on the repository page, you'll see the "Code" button with a dropdown. Click on it to reveal the options.

3. From the dropdown, select "Download ZIP" if you want to download the source code as a ZIP file.
    Alternatively, you can copy the URL shown in the dropdown, which will be in the format: https://github.com/EpicGames/UnrealEngine.git. You can use this URL to clone the repository using Git.
   In your Debian environment, open a terminal and navigate to the directory where you want to clone the repository. Then use the following command to clone the repository:
```shell
git clone https://github.com/EpicGames/UnrealEngine.git
# This will create a local copy of the Unreal Engine source code on your Debian system.
```
Once you have the source code, you can follow the Unreal Engine documentation or build instructions to compile it for your KDE-Debian environment. Make sure to review the system requirements and any additional dependencies needed for building Unreal Engine on Linux.

Note: Building Unreal Engine from source can be a complex process, and it may require substantial resources and time. Make sure you have a compatible Linux distribution, meet the hardware requirements, and carefully follow the instructions provided by Epic Games to successfully build the engine.
